src/main/kotlin/com/amazon/opendistroforelasticsearch/indexmanagement/transform/action/delete/TransportDeleteTransformsAction.kt

```diff
@@ -0,0 +1,127 @@
+/*
+ * Copyright 2021 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License").
+ * You may not use this file except in compliance with the License.
+ * A copy of the License is located at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * or in the "license" file accompanying this file. This file is distributed
+ * on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ * express or implied. See the License for the specific language governing
+ * permissions and limitations under the License.
+ */
+
+package com.amazon.opendistroforelasticsearch.indexmanagement.transform.action.delete
+
+import com.amazon.opendistroforelasticsearch.indexmanagement.IndexManagementPlugin.Companion.INDEX_MANAGEMENT_INDEX
+import org.elasticsearch.ElasticsearchStatusException
+import org.elasticsearch.action.ActionListener
+import org.elasticsearch.action.bulk.BulkRequest
+import org.elasticsearch.action.bulk.BulkResponse
+import org.elasticsearch.action.delete.DeleteRequest
+import org.elasticsearch.action.get.MultiGetRequest
+import org.elasticsearch.action.get.MultiGetResponse
+import org.elasticsearch.action.support.ActionFilters
+import org.elasticsearch.action.support.HandledTransportAction
+import org.elasticsearch.client.Client
+import org.elasticsearch.common.inject.Inject
+import org.elasticsearch.index.IndexNotFoundException
+import org.elasticsearch.rest.RestStatus
+import org.elasticsearch.tasks.Task
+import org.elasticsearch.transport.TransportService
+
+class TransportDeleteTransformsAction @Inject constructor(
+    transportService: TransportService,
+    val client: Client,
+    actionFilters: ActionFilters
+) : HandledTransportAction<DeleteTransformsRequest, BulkResponse>(
+    DeleteTransformsAction.NAME, transportService, actionFilters, ::DeleteTransformsRequest
+) {
+
+    override fun doExecute(task: Task, request: DeleteTransformsRequest, actionListener: ActionListener<BulkResponse>) {
+
+        // TODO: if metadata id exists delete the metadata doc else just delete transform
+
+        // Use Multi-Get Request
+        val getRequest = MultiGetRequest()
+        request.ids.forEach { id ->
+            getRequest.add(MultiGetRequest.Item(INDEX_MANAGEMENT_INDEX, id))
+        }
+
+        client.multiGet(getRequest, object : ActionListener<MultiGetResponse> {
+            override fun onResponse(response: MultiGetResponse) {
+                try {
+                    bulkDelete(response, request.ids, actionListener)
```

Comment:
Can you check if the response is failed before calling `bulkDelete`, you can sample of the responses and see that its failed

```
                    // response is failed only if managed index is not present
                    if (response.responses.first().isFailed) {
                        actionListener.onFailure(
                            ElasticsearchStatusException(
                                "Cluster missing system index $INDEX_MANAGEMENT_INDEX, cannot execute the request", RestStatus.BAD_REQUEST
                            )
                        )
                        return
                    }
```